File: PRACTICA PRIMER PARCIAL/Ejercicio30/Java/NuevaSerie.java
import java.util.Scanner;

public class NuevaSerie {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Lectura de entrada
        System.out.print("Ingrese el número de términos de la serie: ");
        int n = scanner.nextInt();

        // Generar y mostrar la serie
        int[] serie = generarSerie(n);
        for (int i = 0; i < n; i++) {
            System.out.print(serie[i]);
            if (i != n - 1) {
                System.out.print(" ");
            }
        }
    }

    public static int[] generarSerie(int n) {
        int[] serie = new int[n];
        for (int i = 0; i < n; i++) {
            if (i % 3 == 0) {
                serie[i] = 5 + i / 3 * 9;
            } else {
                serie[i] = 3 - i % 3 * 2;
            }
        }
        return serie;
    }
}

File: PRACTICA PRIMER PARCIAL/Ejercicio30/Python/NuevaSerie.py
def generar_serie(n):
    serie = []
    for i in range(n):
        if i % 3 == 0:
            serie.append(5 + i // 3 * 9)
        else:
            serie.append(3 - i % 3 * 2)
    return serie

# Lectura de entrada
n = int(input("Ingrese el número de términos de la serie: "))

# Generar y mostrar la serie
serie = generar_serie(n)
print(" ".join(map(str, serie)))
